### ISSUE TITLE
Health check enabled containers with EBS volume lose volume data after host Power OFF

### ISSUE THREAD

**Issue by soumyalj** [rancher/rancher#14484]
**Rancher versions:**
rancher/server or rancher/rancher:v1.6.19-rc4

**Setup details: (single node rancher vs. HA rancher, internal DB vs. external DB)** : Single node

**Environment Template: (Cattle/Kubernetes/Swarm/Mesos)**: Cattle

**Steps to Reproduce:**
1. Create a setup with 3 hosts "h1, h2, h3" on AWS
2. Launch EBS catalog and create a ebs volume "testvol" from the UI storage tab
3. Create healthcheck enabled service "healthservice" of scale 2 using ebs volume "testvol"
4. Write data to the volume 
5. Power OFF host "h2" where "healthservice" exists.

After Power OFF, one container moved to h1 and another moved to h3.

<img width="842" alt="screen shot 2018-07-10 at 1 02 10 pm" src="https://user-images.githubusercontent.com/18536626/42534450-9bac3a64-8441-11e8-8d07-5a906eb631e0.png">

The container on h2 lost the data
```
root@4ee3e3aefeb1:/# cd /test/
root@4ee3e3aefeb1:/test# ls
ls: reading directory .: Input/output error
root@4ee3e3aefeb1:/test#

```

**Comment by soumyalj** [comment]
In another iteration, it was observed that some of the containers of the healthcheck enabled service were stuck in error state with the below message:
```
 Degraded (Failed to allocate instance [container:1i408]: Bad instance [container:1i408] in state [error]: Allocation failed: SingleHostRW volume is used by mutiple hosts)
```
<img width="1184" alt="screen shot 2018-07-10 at 2 52 28 pm" src="https://user-images.githubusercontent.com/18536626/42539742-e56070b2-8450-11e8-94a0-daca6524bb27.png">

**Comment by StrongMonkey** [comment]
@soumyalj Fix merged, available to test on 1.6 master or new rc.

**Comment by soumyalj** [comment]
Tested with master . I was able to reproduce this issue. 
1. Create a setup with 3 hosts "h1, h2, h3" on AWS
2. Launch EBS catalog and create a ebs volume "testvol" from the UI storage tab
3. Create healthcheck enabled service "healthservice" of scale 10 using ebs volume "testvol"
4. Write data to the volume
5. Power OFF host "h2" where "healthservice" exists.

Two containers moved to host h3 and the volume was Read Only. The data written in step 4 was available for read but further writes not allowed.
```
root@2b20b21d15f7:/# cd /test/
root@2b20b21d15f7:/test# ls
test.txt
root@2b20b21d15f7:/test# echo "Hello again" >test.txt
bash: test.txt: Read-only file system
root@2b20b21d15f7:/test#

```
The other 8 containers on host h1 had the data intact and read/write was allowed .
<img width="1005" alt="screen shot 2018-07-12 at 11 31 48 am" src="https://user-images.githubusercontent.com/18536626/42652742-3000c6a8-85c8-11e8-9061-a4859bf548ca.png">

**Comment by soumyalj** [comment]
Tested with v1.6.19-rc5. I am able to reproduce the issue.
<img width="829" alt="screen shot 2018-07-15 at 1 02 00 pm" src="https://user-images.githubusercontent.com/18536626/42737890-49e32da4-882f-11e8-85a2-450ce2122e44.png">

**Comment by deniseschannon** [comment]
Available in v1.6.19-rc6

**Comment by soumyalj** [comment]
Tested with v1.6.19-rc6. Tested with the steps above several times. The issue did not reproduce.
